engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingproduct/ShortRollingProductOperator.java

```diff
@@ -0,0 +1,136 @@
+import io.deephaven.engine.table.impl.util.RowRedirection;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import static io.deephaven.util.QueryConstants.*;
+
+public class ShortRollingProductOperator extends BaseDoubleUpdateByOperator {
+    private static final int BUFFER_INITIAL_SIZE = 64;
+
+    // region extra-fields
+    // endregion extra-fields
+
+    protected class Context extends BaseDoubleUpdateByOperator.Context {
+        protected ShortChunk<? extends Values> shortInfluencerValuesChunk;
+        protected AggregatingDoubleRingBuffer buffer;
+
+        private int zeroCount;
+
+        protected Context(final int chunkSize) {
+            super(chunkSize);
+            buffer = new AggregatingDoubleRingBuffer(BUFFER_INITIAL_SIZE,
+                    1L,
+                    (a, b) -> a * b, // tree function
+                    (a, b) -> { // value function
+                        if (a == NULL_DOUBLE && b == NULL_DOUBLE) {
+                            return 1L; // identity val
+                        } else if (a == NULL_DOUBLE) {
+                            return b;
+                        } else if (b == NULL_DOUBLE) {
+                            return  a;
+                        }
+                        return a * b;
+                    },
+                    true);
+            zeroCount = 0;
+        }
+
+        @Override
+        public void close() {
+            super.close();
+            buffer = null;
+        }
+
+
+        @Override
+        public void setValuesChunk(@NotNull final Chunk<? extends Values> valuesChunk) {
+            shortInfluencerValuesChunk = valuesChunk.asShortChunk();
+        }
+
+        @Override
+        public void push(int pos, int count) {
+            buffer.ensureRemaining(count);
+
+            for (int ii = 0; ii < count; ii++) {
+                short val = shortInfluencerValuesChunk.get(pos + ii);
+
+                if (val == NULL_SHORT) {
+                    buffer.addUnsafe(NULL_DOUBLE);
+                    nullCount++;
+                } else {
+                    buffer.addUnsafe(val);
+                    if (val == 0) {
+                        zeroCount++;
+                    }
+                }
+            }
+        }
+
+        @Override
+        public void pop(int count) {
+            Assert.geq(buffer.size(), "shortWindowValues.size()", count);
+
+            for (int ii = 0; ii < count; ii++) {
+                double val = buffer.removeUnsafe();
+
+                if (val == NULL_DOUBLE) {
+                    nullCount--;
+                } else if (val == 0) {
+                    --zeroCount;
+                }
+            }
+        }
+
+        @Override
+        public void writeToOutputChunk(int outIdx) {
+            if (buffer.size() == nullCount) {
+                outputValues.set(outIdx, NULL_DOUBLE);
+                curVal = NULL_LONG;
```

Comment:
Should this be `NULL_DOUBLE`? Maybe this is unnecessary as `curVal` is otherwise unused? Or did you intend to  use `curValue` to avoid `evaluate()` calls when possible?

Reply:
Good catch, I removed this assignment.